spec/controllers/tag_wranglers_controller_spec.rb

```diff
@@ -17,4 +17,61 @@
       end
     end
   end
+
+  describe "#report_csv" do
+    let!(:tag1) { create(:freeform, last_wrangler: user, unwrangleable: true) }
+    let!(:tag2) { create(:relationship, last_wrangler: user) }
+
+    context "when anonymous" do
+      it "prevents access" do
+        get :report_csv, params: { id: user.login }
+        it_redirects_to_simple(root_path)
+      end
+    end
+
+    context "when logged in as a superadmin" do
+      let!(:tag3) { create(:relationship, last_wrangler: user, merger: tag2) }
+
+      before { fake_login_admin(create(:superadmin)) }
+
+      it "renders the report" do
+        get :report_csv, params: { id: user.login }
+        result = CSV.parse(response.body.encode("utf-8")[1..], col_sep: "\t")
+        expect(result)
+          .to eq([%w[Name Last\ Updated Type Merger Fandoms Unwrangleable],
```

Comment:
It might also be a good idea to separate the tests for "can download the CSV" from "the CSV contains the right info," so we can more easily tell what the failure is if we ever have one. 

I might be missing it, but I don't think there's a test to make sure the CSV doesn't include tags with a different last wrangler than the one whose CSV is being downloaded.

Reply:
Good idea! I've done the separation and added a test for that case (there was not one to miss originally)